src/MonitorProfiler/ProbeInstrumentation/ProbeInjector.cpp

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "cor.h"
+#include "corprof.h"
+#include "ProbeInjector.h"
+#include "../Utilities/ILRewriter.h"
+
+#include <vector>
+
+//
+// SpecialCaseBoxingTypes shares the same format as other mdTokens.
+// Instrumentation requests provide special boxing instructions by using SpecialCaseBoxingTypeFlag
+// as the "token type" and the SpecialCaseBoxingTypes enum as the RID.
+//
+constexpr ULONG32 SpecialCaseBoxingTypeFlag = 0xff000000;
+enum class SpecialCaseBoxingTypes : ULONG32
+{
+    TYPE_UNKNOWN = 0,
+    TYPE_OBJECT,
+    TYPE_BOOLEAN,
+    TYPE_CHAR,
+    TYPE_SBYTE,
+    TYPE_BYTE,
+    TYPE_INT16,
+    TYPE_UINT16,
+    TYPE_INT32,
+    TYPE_UINT32,
+    TYPE_INT64,
+    TYPE_UINT64,
+    TYPE_SINGLE,
+    TYPE_DOUBLE
+};
+
+HRESULT ProbeInjector::InstallProbe(
+    ICorProfilerInfo* pICorProfilerInfo,
+    ICorProfilerFunctionControl* pICorProfilerFunctionControl,
+    const INSTRUMENTATION_REQUEST& request)
+{
+    IfNullRet(pICorProfilerInfo);
+    IfNullRet(pICorProfilerFunctionControl);
+
+    if (request.boxingTypes.size() > UINT32_MAX)
```

Comment:
Some other possible options here:
- Only process UINT32_MAX entries, ignore the rest
- resize the vector down to UINT32_MAX
- Ignore this condition
- Inject a log entry with a celebratory message